src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,118 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold = 12;
```

Comment:
The `threshold` value should be calculated based on the `DEFAULT_INITIAL_CAPACITY` and `LOAD_FACTOR`, rather than being hardcoded. Consider using `int threshold = (int) (DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);` to ensure that the threshold is always consistent with the capacity and load factor.